src/utils/imageExportUtils.ts

```diff
@@ -0,0 +1,293 @@
+  svg
+    .querySelectorAll(
+      ".tl-fill-hitarea, .tl-stroke-hitarea, .tl-binding-indicator",
+    )
+    .forEach((elm) => elm.remove());
+};
+
+const setBackgroundColor = (svg: SVGElement, app: TldrawApp) => {
+  // Set export background
+
+  const exportBackground: TDExportBackground = app.settings.exportBackground;
+  const darkBackground = "#212529";
+  const lightBackground = "rgb(248, 249, 250)";
+
+  switch (exportBackground) {
+    case TDExportBackground.Auto: {
+      svg.style.setProperty(
+        "background-color",
+        app.settings.isDarkMode ? darkBackground : lightBackground,
+      );
+      break;
+    }
+    case TDExportBackground.Dark: {
+      svg.style.setProperty("background-color", darkBackground);
+      break;
+    }
+    case TDExportBackground.Light: {
+      svg.style.setProperty("background-color", lightBackground);
+      break;
+    }
+    case TDExportBackground.Transparent:
+    default: {
+      svg.style.setProperty("background-color", "transparent");
+      break;
+    }
+  }
+};
+
+const getSvgElementForShape = (
+  shape: TDShape,
+  commonBounds: TLBounds,
+  app: TldrawApp,
+) => {
+  // A quick routine to get an SVG element for each shape
+
+  const util = TLDR.getShapeUtil(shape);
+  const bounds = util.getBounds(shape);
+  const elm = util.getSvgElement(shape, app.settings.isDarkMode);
+
+  if (!elm) return;
+
+  // If the element is an image, set the asset src as the xlinkhref
+  if (shape.type === TDShapeType.Image) {
+    elm.setAttribute("xlink:href", serializeImage(shape.id));
+  } else if (shape.type === TDShapeType.Video) {
+    elm.setAttribute("xlink:href", app.serializeVideo(shape.id));
+  }
+
+  // Put the element in the correct position relative to the common bounds
+  elm.setAttribute(
+    "transform",
+    `translate(${(
+      SVG_EXPORT_PADDING +
+      shape.point[0] -
+      commonBounds.minX
+    ).toFixed(2)}, ${(
+      SVG_EXPORT_PADDING +
+      shape.point[1] -
+      commonBounds.minY
+    ).toFixed(2)}) rotate(${(((shape.rotation || 0) * 180) / Math.PI).toFixed(
+      2,
+    )}, ${(bounds.width / 2).toFixed(2)}, ${(bounds.height / 2).toFixed(2)})`,
+  );
+
+  return elm;
+};
+
+const serializeImage = (id: string): string => {
+  const image = document.getElementById(id + "_image") as HTMLImageElement;
+  if (image) {
+    const canvas = document.createElement("canvas");
+    canvas.width = image.naturalWidth;
+    canvas.height = image.naturalHeight;
+    canvas.getContext("2d")!.drawImage(image, 0, 0);
+    return canvas.toDataURL("image/png");
+  } else throw new Error("Image with id " + id + " not found");
+};
+
+const includeTldrawFonts = async (
+  includeLocalFonts: boolean,
+  style: SVGElement,
+) => {
+  if (!includeLocalFonts) {
+    style.textContent = `@import url('https://fonts.googleapis.com/css2?family=Caveat+Brush&family=Source+Code+Pro&family=Source+Sans+Pro&family=Crimson+Pro&display=block');`;
```

Comment:
Can you please check the CSPs rule in vue if they allow this call?

Reply:
we are using the same csp rules as our vue client
also this doesn't actually import anything, it just sets this import style in svg that is going to be downloaded by user